pkg/beacon/mongodb.go

```diff
@@ -134,6 +135,21 @@ func (r *repo) Upsert(uconnMap map[string]*uconn.Input, hostMap map[string]*host
 
 	// Phase 2: Summary
 
+	// grab the local hosts we have seen during the current analysis period
```

Comment:
In each analysis package, I moved the loop which gathers the local hosts above the initialization code for the summary phase workers. Then I added a check to see if the list of local hosts is empty. If so, then the code prints a message and exits the analysis package early.